optional_plugins/golang/avocado_golang/__init__.py

```diff
@@ -0,0 +1,183 @@
+                                              'src'))
+        if go_path is not None:
+            package_paths.append(os.path.join(os.path.expanduser(go_path),
+                                              'src'))
+
+        if ':' in url:
+            url, _tests_filter = url.split(':', 1)
+            if '/' in _tests_filter:
+                _tests_filter, subtest = _tests_filter.split('/', 1)
+            tests_filter = re.compile(_tests_filter)
+
+        for package_path in package_paths:
+            url_path = os.path.join(package_path, url)
+            files = self._find_files(url_path)
+            if files:
+                test_files.append((package_path, files))
+                break
+
+        if not test_files:
+            if which_tests == loader.ALL:
+                msg = ('No package matching this reference.')
+                return [(NotGolangTest,
+                         {"name": "%s: %s" % (url, msg)})]
+            return []
+
+        for package_path, test_files_list in test_files:
+            for test_file in test_files_list:
+                test_suite = self._find_tests(test_file)
+                for item in test_suite:
+                    common_prefix = os.path.commonprefix([package_path,
+                                                          test_file])
+                    match_package = os.path.relpath(test_file, common_prefix)
+                    test_name = "%s:%s" % (os.path.dirname(match_package),
+                                           item)
+                    if tests_filter and not tests_filter.search(test_name):
+                        continue
+                    avocado_suite.append((GolangTest,
+                                          {'name': test_name,
+                                           'subtest': subtest}))
+        return avocado_suite
+
+    @staticmethod
+    def _find_tests(test_path):
+        test_suite = []
+        with open(test_path, 'r') as test_file_fd:
+            for line in test_file_fd.readlines():
+                if line.startswith('func Test'):
+                    test_suite.append(line.split()[1].split('(')[0])
+
+        return test_suite
+
+    @staticmethod
+    def _find_files(path):
+        matches = []
+        for root, dirnames, filenames in os.walk(path):
+            for filename in fnmatch.filter(filenames, '*_test.go'):
+                matches.append(os.path.join(root, filename))
+
+        return matches
+
+    @staticmethod
+    def get_type_label_mapping():
+        return {GolangTest: 'GOLANG',
+                NotGolangTest: "!GOLANG"}
+
+    @staticmethod
+    def get_decorator_mapping():
+        return {GolangTest: output.TERM_SUPPORT.healthy_str,
+                NotGolangTest: output.TERM_SUPPORT.fail_header_str}
+
+
+class GolangCLI(CLI):
+
+    """
+    Run Golang tests
+    """
+
+    name = 'golang'
+    description = "Golang options for 'run' subcommand"
```

Comment:
This description makes sense from this file point of view, but in `avocado plugins` it doesn't make much sense and yes, there are many descriptions like that, adding card to adjust them (or to discuss whether it's expected, but to me that looks odd).

Reply:
ok, I will create the card.

Reply:
Sorry it's already there... https://trello.com/c/by5KSi69/1145-rfc-review-plugin-descriptions-and-fix-them

Reply:
oh, cool. thanks.